chain/syncer.go

```diff
@@ -189,7 +192,7 @@ func (syncer *Syncer) syncOne(ctx context.Context, grandParent, parent, next typ
 
 	// Run a state transition to validate the tipset and compute
 	// a new state to add to the store.
-	root, err := syncer.stateEvaluator.RunStateTransition(ctx, next, nextMessages, nextReceipts, ancestors, parentWeight, stateRoot)
+	root, err := syncer.stateEvaluator.RunStateTransition(ctx, next, nextBlsMessages, nextSecpMessages, nextReceipts, ancestors, parentWeight, stateRoot)
```

Comment:
Open question about at what point we should validate all the signatures and then strip them and just deal with a single list of messages. I think there are arguments that it could be as high as this point, i.e. immediately after reading them from a block (via the message provider). The BLS message signatures of course must have been already verified - checking all the signatures at the same time, and then assuming them checked, would be simpler in some ways.

We could probably introduce a SigVerifiedMessages wrapper for type checking to help us track that state.

Reply:
This is a great point.  If the validation order were to remain unchanged I am strongly in favor of the argument to do these checks as soon as possible and then deal only with an ordered list of messages across the consensus boundary.  

However because of changes to and ongoing churn in the spec our existing validation pipeline is expected to change a lot.  It's not worth doing this reorganization until we have a solid understanding of the order all of the validation steps at which point we can rearrange everything more or less at once.

Reply:
The question is a good one, but pushing validation higher would change the relationship between syncer and expected (which currently handles all validation). So, not in this PR.